### PR TITLE
Fix/string hash

### PR DESCRIPTION
**Pull request by serge-sans-paille**

### CODE REVIEW

**Comment by serge-sans-paille** [comment]
This is anon-trivial change, any feedback are welcome, I'll wait a bit before merging it.

File: include/frozen/bits/pmh.h [hunk 1]
```diff
@@ -96,13 +96,15 @@ pmh_buckets<M> constexpr make_pmh_buckets(const carray<Item, N> & items,
...
   while (1) {
-    for (auto & b : result.buckets) {
-      b.clear();
-    }
     result.seed = prg();
     for (std::size_t i = 0; i < N; ++i) {
       auto & bucket = result.buckets[hash(key(items[i]), static_cast<size_t>(result.seed)) % M];
-      if (bucket.size() >= result_t::bucket_max) { continue; }
+      if (bucket.size() >= result_t::bucket_max) {
+        continue;
```
Comment (phlptp) [inline include/frozen/bits/pmh.h:103]: The for loop after the continue will never be executed.  Not sure if that was intentional or not.

Comment (serge-sans-paille) [reply include/frozen/bits/pmh.h:103]: mmmh, whaterver the answer, that's unrelated to that PR, right? :-)

Comment (serge-sans-paille) [reply include/frozen/bits/pmh.h:103]: wow my bad, you're 100% correct, I'll fix that.

**Comment by serge-sans-paille** [comment]
Rebased, I'll merge that string hash change tomorrow if no one complains.

**Pull request merged**